st2common/st2common/hooks.py

```diff
@@ -166,8 +167,16 @@ class JSONErrorResponseHook(PecanHook):
 
     def on_error(self, state, e):
         error_msg = getattr(e, 'comment', str(e))
-        LOG.debug('API call failed: %s', error_msg)
-        LOG.debug(traceback.format_exc())
+        extra = {
+            'exception_class': e.__class__.__name__,
+            'exception_message': str(e),
+            'exception_data': e.__dict__
+        }
+
+        LOG.debug('API call failed: %s', error_msg, extra=extra)
+
+        if cfg.CONF.debug:
```

Comment:
I pushed a change so we now log exceptions which result in internal server error using ERROR log level (log.exceptions) - 839d8f16a184cfb5ce571e4f20862aa59a1971d8

Those are unexpected and usually means something is wrong / broken so it should indeed be logged under ERROR.